crates/protoc-gen-grafbase-subgraph/src/main.rs

```diff
@@ -0,0 +1,74 @@
+mod display_utils;
+mod render_graphql_sdl;
+mod schema;
+mod translate_schema;
+
+use prost::Message as _;
+use prost_types::compiler::{CodeGeneratorRequest, CodeGeneratorResponse, code_generator_response::File};
+use render_graphql_sdl::render_graphql_sdl;
+use std::{
+    env,
+    io::{self, Read as _, Write as _},
+    process,
+};
+use translate_schema::translate_schema;
+
+fn bail(error: String) -> CodeGeneratorResponse {
+    CodeGeneratorResponse {
+        error: Some(error),
+        supported_features: None,
+        file: vec![],
+    }
+}
+
+fn generate(raw_request: &[u8]) -> CodeGeneratorResponse {
+    let request = match CodeGeneratorRequest::decode(raw_request) {
+        Ok(request) => request,
+        Err(decode_err) => {
+            return bail(format!(
+                "Failed to decode CodeGeneratorRequest from stdin: {decode_err}",
+            ));
+        }
+    };
+
+    let mut files = Vec::with_capacity(1);
```

Comment:
don't think `with_capacity(1)` is really useful imho, just `vec![file]` would be clearer